### PR TITLE
Transaction tests for dolt, and couple small bug fixes for transactions

### PR DESCRIPTION
**Pull request by zachmu**

### CODE REVIEW

File: go/libraries/doltcore/sqle/dolt_session.go [hunk 1]
```diff
@@ -42,13 +42,15 @@ const (
 
 const EnableTransactionsEnvKey = "DOLT_ENABLE_TRANSACTIONS"
 
-var transactionsEnabled = false
+// TransactionsEnabled controls whether to use SQL transactions
+// Exported only for testing
+var TransactionsEnabled = true
```
Comment (Hydrocharged) [inline go/libraries/doltcore/sqle/dolt_session.go:47]: I think you meant to set this back to `false`. Probably why the test is failing too.

File: go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go [hunk 2]
```diff
@@ -299,3 +299,13 @@ func TestStoredProcedures(t *testing.T) {
 
 	enginetest.TestStoredProcedures(t, newDoltHarness(t))
 }
+
+func TestTransactions(t *testing.T) {
+	if !sqle.TransactionsEnabled {
+		sqle.TransactionsEnabled = true
+		defer func() {
+			sqle.TransactionsEnabled = false
```
Comment (Hydrocharged) [inline go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go:307]: `defer` should probably restore the state rather than force a `false`.

**Review by Hydrocharged — APPROVED** [review]
LGTM!

File: go/libraries/doltcore/sqle/dolt_session.go (hunk 1 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/dolt_session.go:47]: yuuuuup

File: go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go (hunk 2 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go:307]: We only do this if it's false to begin with, so that's equivalent

**Pull request merged**